src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,28 +24,145 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
-    }
 
-    private void refreshTestData(String filename) {
-        testData = loadDataBundle(DATA_FOLDER_PATH + filename);
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            addStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            addStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
         removeAndRestoreDataBundle(testData);
```

Comment:
> Or should each test instead of modifying DataBundle should just read the required number of students, questions and feedback responses from persisted data?

I think it's this one, but will wait for @samsontmr to confirm :)